File: server/entity/backup_profile.go
package entity

import "time"

// BackupProfile defines a backup configuration
type BackupProfile struct {
	ID                uint      `gorm:"primaryKey" json:"id"`
	Name              string    `gorm:"not null" json:"name"`
	ServerID          uint      `gorm:"not null;constraint:OnDelete:RESTRICT" json:"server_id"`
	StorageLocationID uint      `gorm:"not null;constraint:OnDelete:RESTRICT" json:"storage_location_id"`
	NamingRuleID      uint      `gorm:"not null;constraint:OnDelete:RESTRICT" json:"naming_rule_id"`
	ScheduleCron      string    `json:"schedule_cron,omitempty"`
	RetentionDays     *int      `json:"retention_days"` // nil or 0 means keep forever
	Enabled           bool      `json:"enabled"`
	CreatedAt         time.Time `json:"created_at"`

	Server          *Server          `gorm:"foreignKey:ServerID" json:"server,omitempty"`
	StorageLocation *StorageLocation `gorm:"foreignKey:StorageLocationID" json:"storage_location,omitempty"`
	NamingRule      *NamingRule      `gorm:"foreignKey:NamingRuleID" json:"naming_rule,omitempty"`
	Commands        []Command        `gorm:"foreignKey:BackupProfileID;constraint:OnDelete:CASCADE" json:"commands,omitempty"`
	FileRules       []FileRule       `gorm:"foreignKey:BackupProfileID;constraint:OnDelete:CASCADE" json:"file_rules,omitempty"`
	BackupRuns      []BackupRun      `gorm:"foreignKey:BackupProfileID;constraint:OnDelete:CASCADE" json:"backup_runs,omitempty"`
}

File: web/src/types/backup-profile.ts
import type { Server } from './server';
import type { StorageLocation } from './storage-location';
import type { NamingRule } from './naming-rule';
import type { Command } from './command';
import type { FileRule } from './file-rule';
import type { BackupRun } from './backup-run';

export interface BackupProfile {
  id: number;
  name: string;
  server_id: number;
  storage_location_id: number;
  naming_rule_id: number;
  schedule_cron?: string;
  retention_days?: number | null;
  enabled: boolean;
  created_at: string;
  server?: Server;
  storage_location?: StorageLocation;
  naming_rule?: NamingRule;
  commands?: Command[];
  file_rules?: FileRule[];
  backup_runs?: BackupRun[];
}

export interface BackupProfileCreateInput {
  name: string;
  server_id: number;
  storage_location_id: number;
  naming_rule_id: number;
  schedule_cron?: string;
  retention_days?: number | null;
  enabled: boolean;
}

export interface BackupProfileUpdateInput {
  name?: string;
  server_id?: number;
  storage_location_id?: number;
  naming_rule_id?: number;
  schedule_cron?: string;
  retention_days?: number | null;
  enabled?: boolean;
}
